src/modules/sub-items/sub.items.module.js

```diff
@@ -256,12 +267,78 @@ export default class SubItemsModule extends Component {
         this.setState(() => ({ activeView }));
     }
 
+    toggleItemSelect(location, isSelected) {
+        const { selectedLocationsIds } = this.state;
+        const updatedSelectedItemsIds = new Set(selectedLocationsIds);
+
+        if (isSelected) {
+            updatedSelectedItemsIds.add(location.id);
+        } else {
+            updatedSelectedItemsIds.delete(location.id);
+        }
+
+        this.setSelectedItemsState(updatedSelectedItemsIds);
+    }
+
+    toggleAllItemsSelect(isSelectAction) {
+        if (isSelectAction) {
+            this.selectAllItems();
+        } else {
+            this.deselectAllItems();
+        }
+    }
+
+    selectAllItems() {
+        const { items } = this.state;
+        const contentItemsIds = items.map((item) => item.location.id);
+        const contentItemsIdsSet = new Set(contentItemsIds);
+
+        this.setSelectedItemsState(contentItemsIdsSet);
+    }
+
+    deselectAllItems() {
+        this.setSelectedItemsState(new Set());
+    }
+
+    setSelectedItemsState(selectedLocationsIds) {
+        this.setState(() => ({
+            selectedLocationsIds,
+        }));
+    }
+
+    getSelectedItems() {
+        const { items, selectedLocationsIds } = this.state;
+        const selectedLocationsIdsArray = [...selectedLocationsIds];
+        const selectedItems = items.filter((item) => selectedLocationsIdsArray.includes(item.location.id));
+
+        return selectedItems;
+    }
+
+    /**
+     * @param {Set} locationsToRemoveIds
+     */
+    removeItemsFromList(locationsToRemoveIds) {
+        this.setState((state) => {
+            const { items } = state;
+            const filteredItems = items.filter((item) => !locationsToRemoveIds.has(item.location.id));
```

Comment:
`filteredItems` is not a meaningful name.

Reply:
I was thinking for a while for better name and I decided to use `itemsWithoutItemsToRemove`, but it's a bit long, do you have something better in your mind?